### PR TITLE
Improve document and template folder actions

### PR DESCRIPTION
**Pull request by elioschmutz**
Jira: https://4teamwork.atlassian.net/browse/NE-255

This PR adds missing tests for the document and template folder actions and makes some corrections to the exposed template actions:
- `delete`-action will be displayed for documents within dossier templates
- `attach_to_email` will no longer be displayed for templates
- `trash_document` will no longer be displayed for templates
- `send_as_mail` will no longer be displayed within the templates section
- `trash` and `untrash` will no longer be displayed within the templates section

See also: https://4teamwork.atlassian.net/browse/NE-431

## Checklist (Must have)

_Everything has to be done/checked. Checked but not present means the author deemed it unnecessary._

- [x] Changelog entry
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (if applicable)

_Only applicable should be left and checked._

- API change:
  - [ ] Documentation is updated
  - If breaking:
    - [ ] api-change label added
    - [ ] #delivery channel notified about breaking change
    - [ ] Scrum master is informed
- New functionality:
  - [ ] for `document` also works for `mail`
  - [ ] for `task` also works for `forwarding`
- Upgrade steps (changes in profile):
  - [ ] Make it deferrable if possible
  - [ ] Execute as much as possible conditionally
- DB-Schema migration
  - [ ] All changes on a model (columns, etc) are included in a DB-schema migration.
  - [ ] Constraint names are shorter than 30 characters (`Oracle`)
- [ ] Change could impact client installations, client policies need to be adapted
- New translations
  - [ ] All msg-strings are unicode
- Change in schema definition:
  - [ ] If `missing_value` is specified, then `default` has to be set to the same value

### CODE REVIEW

File: opengever/api/tests/test_actions.py [hunk 1]
```diff
@@ -505,6 +505,82 @@ def test_untrashing_available_for_trashed_document(self, browser):
...
+    def test_available_file_actions_for_subdossier_template_document(self, browser):
+        self.login(self.dossier_responsible, browser)
+
+        template = create(Builder('document')
+                          .within(self.subdossiertemplate)
+                          .titled(u'Werkst\xe4tte')
+                          .with_dummy_content())
+
+        expected_file_actions = [
+            {u'icon': u'', u'id': u'oc_direct_checkout', u'title': u'Checkout and edit'},
```
Comment (njohner) [inline opengever/api/tests/test_actions.py:550]: hum why is that different from `dossier_template_document`??

File: opengever/document/fileactions.py [hunk 2]
```diff
@@ -99,7 +99,10 @@ def is_revert_to_version_action_available(self):
 
     def is_trash_document_available(self):
         trasher = ITrashable(self.context)
-        return trasher.verify_may_trash(raise_on_violations=False)
+        return (
+            trasher.verify_may_trash(raise_on_violations=False)
```
Comment (njohner) [inline opengever/document/fileactions.py:103]: Shouldn't we rather add that check in the `trasher.verify_may_trash` method?

**Review by njohner — COMMENTED** [review]
LGTM, just some minor comments

File: opengever/api/tests/test_actions.py (hunk 1 above)
Comment (elioschmutz) [reply opengever/api/tests/test_actions.py:591]: 👍 Updated. It was because I used a different user

File: opengever/document/fileactions.py (hunk 2 above)
Comment (elioschmutz) [reply opengever/document/fileactions.py:103]: I'm not sure. It's possible to trash. But it's not available for this object. So I think, it belongs to this file...

**Comment by elioschmutz** [comment]
@njohner I updated the PR and I also added some more commits to also update the template folder actions. Could you take another look?

**Review by njohner — APPROVED** [review]
LGTM

**Comment by elioschmutz** [comment]
@njohner FYI, I added a new commit to also remove `attach_documents` in the templates-area.

**Review by njohner — APPROVED** [review]
LGTM

**Pull request merged**